### PR TITLE
PP-4594 Add last logged in at property

### PR DESCRIPTION
**Pull request by DanailMinchev**
## WHAT

- Added `lastLoggedInAt` property to `UserEntity`
- Added `lastLoggedInAt` property to `User`

### CODE REVIEW

File: src/main/java/uk/gov/pay/adminusers/service/UserServices.java [hunk 1]
```diff
@@ -173,7 +172,7 @@ public UserServices(UserDao userDao,
     }
 
     @Transactional
-    public Optional<User> authenticateSecondFactor(String externalId, int code) {
+    public Optional<User> authenticateSecondFactor(String externalId, int code, ZonedDateTime now) {
```
Comment (SandorArpa) [inline src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: Why are we passing in the value instead of initialising it locally?

Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: Initially I initialised it locally, but later realised that to be able to test it, we need to mock `ZonedDateTime.now` for example here https://github.com/alphagov/pay-adminusers/blob/e9f7ce496bf24d14860206be439b5a50213a4a3d/src/test/java/uk/gov/pay/adminusers/service/UserServicesTest.java#L437 and then assert on a mocked value.
We cannot mock static methods without introducing something like https://github.com/powermock/powermock (at least I was unable to find other solution with quick research).

That is why I have decided to move the value as part of the method call to be able to test it properly.

Please let me know if you have any other ideas how to test that, thanks

Comment (SandorArpa) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: You could test it with a relative value, eg:

`ZonedDateTime now = ZonedDateTime.now().minusSeconds(10);
...
 assertThat(user.getLastLoggedInAt().isAfter(now), is(true));`

Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: Good point! I will try that

Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: As discussed with @alexbishop1 and @SandorArpa we currently use same approach as @SandorArpa suggested using https://github.com/alphagov/pay-connector/blob/master/src/test/java/uk/gov/pay/connector/matcher/ZoneDateTimeAsStringWithinMatcher.java#L24

Also, we discussed that ideally we should inject `Clock` class into the service and then manipulate it for testing as suggested by @alexbishop1 but we agreed that it is not the current pattern which we use, so we will not do that for now

File: src/main/java/uk/gov/pay/adminusers/service/UserServices.java [hunk 2]
```diff
@@ -175,6 +174,7 @@ public UserServices(UserDao userDao,
     @Transactional
     public Optional<User> authenticateSecondFactor(String externalId, int code) {
         logger.debug("OTP attempt - user_id={}", externalId);
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC"));
```
Comment (SandorArpa) [inline src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: You could also use `ZonedDateTime.now(ZoneOffset.UTC)`

Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java]: Yes, looks like we use `ZonedDateTime.now(ZoneId.of("UTC"))` a bit more in the project

**Pull request merged**